CI/release.py

```diff
@@ -140,8 +145,98 @@ def markdown_changelog(version: str, changelog: dict, header: bool = False) -> s
     return output
 
 
-async def main(draft, dry_run):
-    token = os.environ["GH_TOKEN"]
+def update_zenodo(zenodo_file: Path, repo: str, next_version):
+    data = json.loads(zenodo_file.read_text())
+    data["title"] = f"{repo}: v{next_version}"
+    data["version"] = f"v{next_version}"
+    zenodo_file.write_text(json.dumps(data, indent=2))
+
+
+def update_zenodo(zenodo_file: Path, repo: str, next_version):
+    data = json.loads(zenodo_file.read_text())
+    data["title"] = f"{repo}: v{next_version}"
+    data["version"] = f"v{next_version}"
+    zenodo_file.write_text(json.dumps(data, indent=2))
+
+
+def update_citation(citation_file: Path, next_version):
+    with citation_file.open() as fh:
+        data = yaml.safe_load(fh)
+    data["version"] = f"v{next_version}"
+    data["date-released"] = datetime.date.today().strftime("%Y-%m-%d")
+    with citation_file.open("w") as fh:
+        yaml.dump(data, fh, indent=2)
+
+
+def make_sync(fn):
+    @functools.wraps(fn)
+    def wrapped(*args, **kwargs):
+        loop = asyncio.get_event_loop()
+        loop.run_until_complete(fn(*args, **kwargs))
+
+    return wrapped
+
+
+app = typer.Typer()
+
+
+async def get_parsed_commit_range(
+    start: str, end: str, repo: str, gh: GitHubAPI, edit: bool = False
+) -> Tuple[List[Commit], List[Commit]]:
+    commits_iter = gh.getiter(f"/repos/{repo}/commits?sha={start}")
+
+    commits = []
+    unparsed_commits = []
+
+    try:
+        async for item in commits_iter:
+            commit_hash = item["sha"]
+            commit_message = item["commit"]["message"]
+            if commit_hash == end:
+                break
+
+            invalid_message = False
+            try:
+                _default_parser(commit_message)
+                # if this succeeds, do nothing
+            except UnknownCommitMessageStyleError as err:
+                print("Unknown commit message style!")
+                if not commit_message.startswith("Merge"):
+                    invalid_message = True
+            if (
+                (invalid_message or edit)
+                and sys.stdout.isatty()
+                and False
+                and typer.confirm(f"Edit effective message '{commit_message}'?")
+            ):
+                commit_message = typer.edit(commit_message)
+                _default_parser(commit_message)
+
+            commit = Commit(commit_hash, commit_message)
+            commits.append(commit)
+
+            if invalid_message:
+                unparsed_commits.append(commit)
+
+            print("-", commit)
+            if len(commits) > 100:
+                raise RuntimeError(f"{len(commits)} are a lot. Aborting!")
```

Comment:
That's just a failsafe in case something goes wrong with the merge commit and it never finds the "previous version" commit. I would rather not have it spit out an incorrect giant changelog than tell me about it here.

Reply:
I actually increased it to 200, to lower the possibility of a legitimate release crossing the threshold.

Reply:
Makes sense, our biggest releases so far were around 50 commits so 200 looks like a large enough safety factor.